stan/math/prim/fun/to_ref.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef STAN_MATH_PRIM_FUN_TO_REF_HPP
+#define STAN_MATH_PRIM_FUN_TO_REF_HPP
+
+#include <stan/math/prim/meta.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * No-op that should be optimized away.
+ * @tparam T non-Eigen argument type
+ * @param a argument
+ * @return argument
+ */
+template <typename T, require_not_eigen_t<T>* = nullptr>
+inline T to_ref(T&& a) {
```

Comment:
I think not. For rvalue arguments this returns value type, but we are relying on compiler to do RVO, so hpefully no copy happens. 

`decltype(auto)` would (I think) make return type `T&&`. That is the function would for rvalue arguments return a rvalue reference. I think if the return value is stored into a reference variable the rvalue could be destroyed, leaving us with a dangling reference.

I am not 100% sure in which cases this can happen, so I prefer staying on the safe side and relying on RVO. I you have better understanding of this, please explain.

Reply:
RVO sounds good to me, still working on understanding the nuts and bolts of forwarding, so I have no strong preferences here!